django_project/lesson/tests/test_worksheet_views.py

```diff
@@ -0,0 +1,144 @@
+                    'project_slug': self.test_section.project.slug,
+                    'section_slug': self.test_section.slug}),
+                post_data)
+        self.assertEqual(response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetCreateView_with_login(self):
+        """Test accessing worksheet create view with login."""
+
+        status = self.client.login(username = 'sonlinux', password =
+        'password')
+        self.assertTrue(status)
+        post_data = {
+            'module': u'Demo worksheet name',
+            'title': u'Demo worksheet title',
+            'section': self.test_section.id,
+        }
+        response = self.client.post(
+                reverse('worksheet-create', kwargs = {
+                    'project_slug': self.test_section.project.slug,
+                    'section_slug': self.test_section.slug}),
+                post_data)
+
+        self.assertEqual(response.status_code, 200)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetDetailView(self):
+        """Tests accessing worksheet detail view."""
+
+        response = self.client.get(reverse(
+                'worksheet-detail', kwargs = self.kwargs_worksheet_full))
+        self.assertEqual(response.status_code, 200)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetUpdateView(self):
+        """Tests updating worksheet without login."""
+
+        response = self.client.get(reverse(
+                'worksheet-update', kwargs = self.kwargs_worksheet_full))
+        self.assertEqual(response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetUpdateView_with_login(self):
+        """Tests updating worksheet with login."""
+
+        status = self.client.login(username='sonlinux', password='password')
+        self.assertTrue(status)
+        response = self.client.get(reverse(
+            'worksheet-update', kwargs=self.kwargs_worksheet_full))
+        self.assertEqual(response.status_code, 200)
+        expected_templates = [
+            'update.html'
+        ]
+        self.assertEqual(response.template_name, expected_templates)
+
+    @override_settings(VALID_DOMAIN=['testserver', ])
+    def test_WorksheetModuleQuestionAnswers(self):
+
+        response = self.client.get(reverse('worksheet-module-answers',
+                                           kwargs=self.kwargs_worksheet_full))
+        self.assertEqual(response.status_code, 302)
+
+    @override_settings(VALID_DOMAIN = ['testserver', ])
+    def test_WorksheetModuleQuestionAnswers_with_login(self):
+
+        status = self.client.login(username = 'sonlinux',
+                                   password = 'password')
+        self.assertTrue(status)
+
+        response = self.client.get(reverse('worksheet-module-answers', kwargs =
+        self.kwargs_worksheet_full))
```

Comment:
__[Flake8]__

[E128] continuation line under-indented for visual indent


<!-- Comment by [Sider](https://sider.review) -->